addons/block_code/ui/main_panel.gd

```diff
@@ -85,6 +85,9 @@ func _on_delete_node_button_pressed():
 	EditorInterface.popup_dialog_centered(dialog)
 	dialog.connect("confirmed", _on_delete_dialog_confirmed.bind(_context.block_code_node))
 
+func _on_advanced_checkbox_toggled(is_advanced: bool):
+	_picker.reload_blocks(is_advanced)
```

Comment:
There is another call to `_picker.reload_blocks()` in `main_panel.gd` . You should pass `is_advanced` from the UI checkbox to this one too. Otherwise there is a bug:
- Turn "is advanced" on.
- See the advanced blocks appear in the block picker.
- Add a new Variable.
- See the advanced blocks dissappear, even if the "is advanced" is turned on.
[Grabación de pantalla desde 2024-12-16 10-14-12.webm](https://github.com/user-attachments/assets/670140b4-c8c8-4a22-8907-028802da460b)

Reply:
Good catch @manuq , I fixed this in 5a570c0 by adding `_advanced` to Picker variable instead of passing it as a parameter into `reload_blocks`